### PR TITLE
config from env (with .env support)

### PR DESCRIPTION
**Pull request by nl0**
## Summary

Get config values from env (overriding with `.env` file supported), remove `static/config.js`;

### Current solution

production (docker-based) and development environments use different files as the source of truth for configuration:
* prod uses `config.js.tmpl` with substitution of env vars
* dev uses `static/config.js` with hardcoded values

### Problems with the current solution

1. Different sources of truth for different environments: you should always remember to keep these sources in sync (see [this chapter](https://12factor.net/config) in The Twelve-Factor App)

2. Discomfort when developing: one must edit the config file and then make sure it doesn't get committed into the repo, so we have these dances with stashing the changes to the config or commenting / uncommenting lines. It's not that bad, but kinda annoying.

### Proposed solution

Use env vars and a single config template file as the source of truth. We already have the config template, so all we need to do is to remove the duplicate config with hardcoded values and serve the template with substituted values in the dev mode, which is trivial. This solves the problems mentioned above. The only drawback I see here is that there's no defaults now: all the values required by the config must be provided by the env, but it's solved easily with introduction of `.env` files.

### CODE REVIEW

File: catalog/.env.example [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Example env file: save as .env (ignored by git) and edit to use.
```
[the remark this answers is not in the document]
Comment (akarve) [reply catalog/.env.example]: just to be 100% explicit, you _do not_ expect the Python code (e.g. Flask) to pick up .env, but you do expect the app to use the same config file for front and back end (config.js.tmpl)?

[the remark this answers is not in the document]
Comment (akarve) [reply catalog/.env.example]: i suggest at least a line and a link to this logic in /catalog/README.md

[the remark this answers is not in the document]
Comment (nl0) [reply catalog/.env.example]: > just to be 100% explicit, you do not expect the Python code (e.g. Flask) to pick up .env, but you do expect the app to use the same config file for front and back end (config.js.tmpl)?

that's correct

[the remark this answers is not in the document]
Comment (nl0) [reply catalog/.env.example]: > i suggest at least a line and a link to this logic in /catalog/README.md

sure, i'll update the readme

File: catalog/.env.example [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+# Registry (api) url
+# local
+REGISTRY_URL=http://flask:5000
+# staging registry instance
```
[the remark this answers is not in the document]
Comment (nl0) [reply catalog/.env.example]: will do

File: catalog/.env.example [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#REGISTRY_URL=https://stage-registry.quiltdata.com
```
[the remark this answers is not in the document]
Comment (nl0) [reply catalog/.env.example]: the last one wins, yes

**Comment by nl0** [comment]
@akarve updated

**Pull request merged**